### PR TITLE
[BE] Simplify tests, elaborate testnames in test_optim.py

### PR DESCRIPTION
**Pull request by janeyx99**
- Deletes unused kwargs
- Make test names more descriptive to remove need of comments. Overall it's better to codify over comment
- Added a test for duplicate params across groups
- Greatly simplified test_empty_grad to discover that the crux of the bug was NOT its emptiness, but rather with multi-dim emptiness.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #101103
* #101100
* __->__ #101004

### CODE REVIEW

File: test/test_optim.py [hunk 1]
```diff
@@ -1500,9 +1498,6 @@ def test_asgd(self):
                     [params], maximize=False, weight_decay=0.9, foreach=foreach
                 )
             )
-            self._test_complex_optimizer(
```
Comment (albanD) [inline test/test_optim.py:1503]: This is a duplicate?

File: test/test_optim.py [hunk 2]
```diff
@@ -1568,41 +1563,93 @@ def closure():
...
+        )
 
-    def test_duplicate_params_in_param_group(self):
-        param = Parameter(torch.randn(5, 5))
+    def test_duplicate_params_in_one_param_group(self):
+        param = Parameter(torch.randn(1))
         with warnings.catch_warnings(record=True) as w:
             warnings.simplefilter("always")
-            optim.SGD([param, param], lr=0.1)
+            optim.SGD([param, param], lr=0.01)
```
Comment (albanD) [inline test/test_optim.py:1576]: nit: `self.assertWarnsRegex()`

**Review by albanD — COMMENTED** [review]
Sounds pretty good!

File: test/test_optim.py (hunk 2 above)
Comment (janeyx99) [reply test/test_optim.py:1576]: :o til

**Review by albanD — APPROVED** [review]
Ok!

**Pull request closed without merging**